source/common/http/http2/codec_impl.cc

```diff
@@ -1431,6 +1433,52 @@ void ConnectionImpl::dumpState(std::ostream& os, int indent_level) const {
   }
 }
 
+void ConnectionImpl::dumpStreams(std::ostream& os, int indent_level) const {
+  const char* spaces = spacesForLevel(indent_level);
+
+  // Try to dump details for the current stream.
+  // If none, dump a subset of our active streams.
+  os << spaces << "Number of active streams: " << active_streams_.size()
+     << DUMP_OPTIONAL_MEMBER(current_stream_id_);
+
+  if (current_stream_id_.has_value()) {
+    os << " Dumping current stream:\n";
+    const ConnectionImpl::StreamImpl* stream = getConstStream(current_stream_id_.value());
+    DUMP_DETAILS(stream);
+  } else {
+    os << " Dumping " << std::min<size_t>(25, active_streams_.size()) << " Active Streams:\n";
+    size_t count = 0;
+    for (auto& stream : active_streams_) {
+      DUMP_DETAILS(stream);
+      if (++count >= 25) {
+        break;
+      }
+    }
+  }
+}
+
+void ClientConnectionImpl::dumpStreams(std::ostream& os, int indent_level) const {
+  ConnectionImpl::dumpStreams(os, indent_level);
+
+  if (!current_stream_id_.has_value()) {
+    return;
+  }
+
+  // Try to dump the downstream request information, corresponding to the
+  // stream we were processing.
+  const char* spaces = spacesForLevel(indent_level);
+  os << spaces << "Dumping corresponding downstream request for upstream stream "
+     << current_stream_id_.value() << ":\n";
+
+  const ClientStreamImpl* client_stream =
+      static_cast<const ClientStreamImpl*>(getConstStream(current_stream_id_.value()));
+  if (client_stream) {
+    client_stream->response_decoder_.dumpState(os, indent_level + 1);
+  } else {
+    os << spaces << " Failed to get current stream.\n";
```

Comment:
Maybe be clearer about what's happening here? Ie that we're processing a callback for an unknown stream

Reply:
Added a bit more to the logging.